### ISSUE TITLE
CoNLL training data converter for skills NER

### ISSUE THREAD

**Issue by slavb18** [iconicompany/iskillmatching#2]
Адаптировать и перенести обучение модели из https://github.com/iconicompany/NER-Web-App-TensorFlowJS/pull/1

The existing pipeline trains on CoNLL 2003 data; there was no path to train on a domain-specific skills dictionary (one skill per line, e.g. `React`, `.Net`, `Java`).

### `data/prepare_skills_data.py` (new)
Converts a `skills.csv` file into CoNLL-format train/valid/test splits consumable by the existing `model_tf/model_tuner.py` pipeline without any model changes.

- Reads skills (one per line); no limit on vocabulary size
- Generates synthetic sentences via 20 context templates per skill (`"Experience with {skill} is required."`, etc.) to provide realistic token context
- Pre-processes each sentence through the same `TextPreprocessor` the model uses at inference — tokenisation is identical at both stages
- Finds skill tokens via subsequence search and assigns BIO tags; punctuation-heavy skills are handled correctly:
  - `.Net` → `. Net` → `B-SKILL I-SKILL`
  - `C++` → `C + +` → `B-SKILL I-SKILL I-SKILL`
  - `Spring Boot` → `B-SKILL I-SKILL`
- Adds O-only negative examples to prevent the model from over-labelling
- Shuffles and writes `train.txt` / `valid.txt` / `test.txt`; split ratios are configurable

```bash
python data/prepare_skills_data.py \
    --skills data/skills.csv \
    --output_dir data/skills \
    --train_ratio 0.7 --valid_ratio 0.15
```

After generation, point `config.json` at the new files and re-run the notebook data-preparation cells (`CustomTokenizer` fit, `SequentialLabelEncoder` fit, `data.joblib` serialisation) before training. `NUM_CLASSES` will update to 3 (`O`, `B-SKILL`, `I-SKILL`).

### `data/skills.csv` (new)
Sample dictionary of 69 common tech skills (React, .Net, Java, Python, Docker, Kubernetes, …) for testing and as a template.

### `data/skills/` (new)
Pre-generated example CoNLL output from the sample dictionary so the expected format is immediately visible.
